packages/account/src/account.ts

```diff
@@ -500,6 +502,45 @@ export class Account extends AbstractAccount {
     return this.sendTransaction(request);
   }
 
+  async getTransactionCost(
+    transactionRequestLike: TransactionRequestLike,
+    { signatureCallback, quantitiesToContract = [] }: TransactionCostParams = {}
+  ): Promise<TransactionCost> {
+    const txRequestClone = clone(transactionRequestify(transactionRequestLike));
+    const isScriptTransaction = txRequestClone.type === TransactionType.Script;
+    const baseAssetId = this.provider.getBaseAssetId();
+
+    // Fund with fake UTXOs to avoid not enough funds error
+    // Getting coin quantities from amounts being transferred
+    const coinOutputsQuantities = txRequestClone.getCoinOutputsQuantities();
+    // Combining coin quantities from amounts being transferred and forwarding to contracts
+    const requiredQuantities = mergeQuantities(coinOutputsQuantities, quantitiesToContract);
+    const resources = this.generateFakeResources(
+      mergeQuantities(
+        [...requiredQuantities],
+        [{ assetId: baseAssetId, amount: bn('100000000000000000') }]
+      )
+    );
+    txRequestClone.addResources(resources);
+
+    /**
+     * Estimate predicates gasUsed
+     */
+    // Remove gasLimit to avoid gasLimit when estimating predicates
+    if (isScriptTransaction) {
+      txRequestClone.gasLimit = bn(0);
+    }
```

Comment:
@petertonysmith94 Perhaps this should be addressed on `Provider.getTransactionCost`

Reply:
f6f66789fc41f704493de07a6f125d1de69f2f40